src/Workspaces/CoreTest/FunctionIdTests.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.CodeAnalysis.Internal.Log;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.UnitTests
+{
+    public class FunctionIdTests
+    {
+        [Fact]
+        public void NoDuplicateValues()
+        {
+            var map = new Dictionary<FunctionId, string>();
+            foreach (var name in Enum.GetNames(typeof(FunctionId)))
+            {
+                var value = (FunctionId)Enum.Parse(typeof(FunctionId), name);
+                if (map.TryGetValue(value, out var existingName))
+                {
+                    Assert.True(false, $"'{nameof(FunctionId)}.{name}' cannot have the same value as '{nameof(FunctionId)}.{existingName}'");
```

Comment:
```suggestion
                    throw new Exception($"'{nameof(FunctionId)}.{name}' cannot have the same value as '{nameof(FunctionId)}.{existingName}'");
```

Otherwise you get a lot of head scratching as to why we expected true and got false.

Reply:
This is the official xUnit pattern for `Assert.Fail`.
https://xunit.net/docs/comparisons.html

Reply:
Oh really I had always thought the advice was indeed just throw.